app/src/main/java/com/example/drawingactivity/ColorUtils.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.drawingactivity
```

Comment:
interesting approach  of separating the color based functions from the sheet. I ended up putting these kind of functions in my viewModel, because that's where I was storing and getting the values from and thought it was viewModel logic because it was handling the state and being told the state changed. I imagine in your case your handling everything in the view so everything is instead viewlogic and should be handled as such.

Reply:
Yep! I was using the viewModel only to hold the last HSV value and everything else is done on a view level since only the gradients and edit texts need to handle that